scripts/redeploy.sh

```diff
@@ -6,10 +6,10 @@ echo "] Removing the war file ..."
 sudo rm -rf /var/lib/tomcat7/webapps/protocolanalyzer.war /var/lib/tomcat7/webapps/protocolanalyzer
 echo "] Remove complete."
 echo ""
-echo "] Restarting tomcat7 ..."
-sudo service tomcat7 restart
+echo "] Restarting tomcat ..."
+sudo service tomcat restart
 echo "] Restart complete."
 echo ""
 echo "] Deploying Application ..."
-mvn tomcat7:deploy
-echo "] Deploy complete"
\ No newline at end of file
+mvn tomcat:deploy
```

Comment:
This command is failing. I tried with the existing tomcat7-maven-plugin and also cargo-maven2-plugin. Both are failing for the same reason. The problem may be in our deployment configuration. The failure log is given below.

```
[INFO] Deploying war to http://localhost:8080/protocolanalyzer  
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 05:00 min
[INFO] Finished at: 2017-11-17T09:07:47+05:30
[INFO] Final Memory: 21M/201M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:tomcat-maven-plugin:1.1:deploy (default-cli) on project protocolanalyzer: Cannot invoke Tomcat manager: Error writing to server -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```
A more detailed debug log of the same problem is given below.
```
[INFO] <<< tomcat-maven-plugin:1.1:deploy (default-cli) < package @ protocolanalyzer <<<
[INFO] 
[INFO] --- tomcat-maven-plugin:1.1:deploy (default-cli) @ protocolanalyzer ---
[DEBUG] Dependency collection stats: {ConflictMarker.analyzeTime=0, ConflictMarker.markTime=0, ConflictMarker.nodeCount=97, ConflictIdSorter.graphTime=0, ConflictIdSorter.topsortTime=0, ConflictIdSorter.conflictIdCount=32, ConflictIdSorter.conflictIdCycleCount=0, ConflictResolver.totalTime=1, ConflictResolver.conflictItemCount=75, DefaultDependencyCollector.collectTime=36, DefaultDependencyCollector.transformTime=1}
[DEBUG] org.codehaus.mojo:tomcat-maven-plugin:jar:1.1:
[DEBUG]    org.apache.maven:maven-plugin-api:jar:2.0.8:compile
[DEBUG]    org.apache.maven:maven-artifact-manager:jar:2.0.8:compile
[DEBUG]       org.apache.maven:maven-repository-metadata:jar:2.0.8:compile
[DEBUG]       org.codehaus.plexus:plexus-container-default:jar:1.0-alpha-9-stable-1:compile
[DEBUG]          junit:junit:jar:3.8.2:test
[DEBUG]          classworlds:classworlds:jar:1.1-alpha-2:compile
[DEBUG]    org.apache.maven.wagon:wagon-provider-api:jar:1.0-alpha-5:compile
[DEBUG]    org.apache.maven:maven-artifact:jar:2.0.8:compile
[DEBUG]    org.apache.maven:maven-project:jar:2.0.8:compile
[DEBUG]       org.apache.maven:maven-settings:jar:2.0.8:compile
[DEBUG]       org.apache.maven:maven-profile:jar:2.0.8:compile
[DEBUG]       org.apache.maven:maven-model:jar:2.0.8:compile
[DEBUG]       org.apache.maven:maven-plugin-registry:jar:2.0.8:compile
[DEBUG]    org.apache.maven:maven-archiver:jar:2.2:compile
[DEBUG]       org.codehaus.plexus:plexus-archiver:jar:1.0-alpha-7:compile
[DEBUG]    org.codehaus.plexus:plexus-utils:jar:1.0.4:compile
[DEBUG]    commons-codec:commons-codec:jar:1.3:compile
[DEBUG]    org.codehaus.plexus:plexus-classworlds:jar:2.2.2:compile
[DEBUG]    org.apache.tomcat:catalina:jar:6.0.29:compile
[DEBUG]       org.apache.tomcat:juli:jar:6.0.29:compile
[DEBUG]       org.apache.tomcat:annotations-api:jar:6.0.29:compile
[DEBUG]    org.apache.tomcat:catalina-ha:jar:6.0.29:runtime
[DEBUG]    org.apache.tomcat:tribes:jar:6.0.29:runtime
[DEBUG]    org.apache.tomcat:el-api:jar:6.0.29:runtime
[DEBUG]    org.apache.tomcat:jasper:jar:6.0.29:runtime
[DEBUG]    org.apache.tomcat:jasper-el:jar:6.0.29:runtime
[DEBUG]    org.apache.tomcat:jasper-jdt:jar:6.0.29:runtime
[DEBUG]    org.apache.tomcat:jsp-api:jar:6.0.29:runtime
[DEBUG]    org.apache.tomcat:servlet-api:jar:6.0.29:runtime
[DEBUG]    org.apache.tomcat:coyote:jar:6.0.29:runtime
[DEBUG]    org.apache.tomcat:dbcp:jar:6.0.29:runtime
[DEBUG] Created new class realm plugin>org.codehaus.mojo:tomcat-maven-plugin:1.1
[DEBUG] Importing foreign packages into class realm plugin>org.codehaus.mojo:tomcat-maven-plugin:1.1
[DEBUG]   Imported:  < maven.api
[DEBUG] Populating class realm plugin>org.codehaus.mojo:tomcat-maven-plugin:1.1
[DEBUG]   Included: org.codehaus.mojo:tomcat-maven-plugin:jar:1.1
[DEBUG]   Included: org.apache.maven:maven-archiver:jar:2.2
[DEBUG]   Included: org.codehaus.plexus:plexus-archiver:jar:1.0-alpha-7
[DEBUG]   Included: org.codehaus.plexus:plexus-utils:jar:1.0.4
[DEBUG]   Included: commons-codec:commons-codec:jar:1.3
[DEBUG]   Included: org.apache.tomcat:catalina:jar:6.0.29
[DEBUG]   Included: org.apache.tomcat:juli:jar:6.0.29
[DEBUG]   Included: org.apache.tomcat:annotations-api:jar:6.0.29
[DEBUG]   Included: org.apache.tomcat:catalina-ha:jar:6.0.29
[DEBUG]   Included: org.apache.tomcat:tribes:jar:6.0.29
[DEBUG]   Included: org.apache.tomcat:el-api:jar:6.0.29
[DEBUG]   Included: org.apache.tomcat:jasper:jar:6.0.29
[DEBUG]   Included: org.apache.tomcat:jasper-el:jar:6.0.29
[DEBUG]   Included: org.apache.tomcat:jasper-jdt:jar:6.0.29
[DEBUG]   Included: org.apache.tomcat:jsp-api:jar:6.0.29
[DEBUG]   Included: org.apache.tomcat:servlet-api:jar:6.0.29
[DEBUG]   Included: org.apache.tomcat:coyote:jar:6.0.29
[DEBUG]   Included: org.apache.tomcat:dbcp:jar:6.0.29
[DEBUG] Configuring mojo org.codehaus.mojo:tomcat-maven-plugin:1.1:deploy from plugin realm ClassRealm[plugin>org.codehaus.mojo:tomcat-maven-plugin:1.1, parent: sun.misc.Launcher$AppClassLoader@5c647e05]
[DEBUG] Configuring mojo 'org.codehaus.mojo:tomcat-maven-plugin:1.1:deploy' with basic configurator -->
[DEBUG]   (f) charset = ISO-8859-1
[DEBUG]   (f) contextFile = /home/darshini/darshini-install/target/protocolanalyzer-1.0-SNAPSHOT/META-INF/context.xml
[DEBUG]   (f) ignorePackaging = false
[DEBUG]   (f) mode = war
[DEBUG]   (f) packaging = war
[DEBUG]   (f) path = /protocolanalyzer
[DEBUG]   (f) update = false
[DEBUG]   (f) url = http://localhost:8080/manager
[DEBUG]   (f) version = 1.1
[DEBUG]   (f) warFile = /home/darshini/darshini-install/target/protocolanalyzer-1.0-SNAPSHOT.war
[DEBUG] -- end configuration --
[INFO] Deploying war to http://localhost:8080/protocolanalyzer  
[DEBUG] No server specified for authentication - using defaults
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3.102 s
[INFO] Finished at: 2017-11-17T09:08:50+05:30
[INFO] Final Memory: 23M/376M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:tomcat-maven-plugin:1.1:deploy (default-cli) on project protocolanalyzer: Cannot invoke Tomcat manager: Error writing to server -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.codehaus.mojo:tomcat-maven-plugin:1.1:deploy (default-cli) on project protocolanalyzer: Cannot invoke Tomcat manager
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Cannot invoke Tomcat manager
	at org.codehaus.mojo.tomcat.AbstractCatalinaMojo.execute(AbstractCatalinaMojo.java:149)
	at org.codehaus.mojo.tomcat.AbstractWarCatalinaMojo.execute(AbstractWarCatalinaMojo.java:70)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 20 more
Caused by: java.io.IOException: Error writing to server
	at sun.net.www.protocol.http.HttpURLConnection.writeRequests(HttpURLConnection.java:699)
	at sun.net.www.protocol.http.HttpURLConnection.writeRequests(HttpURLConnection.java:711)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1585)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
	at org.codehaus.mojo.tomcat.TomcatManager.invoke(TomcatManager.java:604)
	at org.codehaus.mojo.tomcat.TomcatManager.deployImpl(TomcatManager.java:662)
	at org.codehaus.mojo.tomcat.TomcatManager.deploy(TomcatManager.java:295)
	at org.codehaus.mojo.tomcat.AbstractDeployWarMojo.deployWar(AbstractDeployWarMojo.java:85)
	at org.codehaus.mojo.tomcat.AbstractDeployMojo.invokeManager(AbstractDeployMojo.java:85)
	at org.codehaus.mojo.tomcat.AbstractCatalinaMojo.execute(AbstractCatalinaMojo.java:141)
	... 23 more
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
] Deploy complete
```